### PR TITLE
[CDAP-7063][CDAP-7251] - Initial implementation of cdap in react

### PR DESCRIPTION
**Pull request by ajainarayanan**
- Adds the initial version of cdap in React.
- Moves Hydrator and Tracker as separate angular app.
- Adds initial version of webpack.
- Borrows common styling from old cdap for start
#### Things to do:
- Single signon
- Figure out how to build hydrator, tracker and cdap simultaneously with one command.
- Figure out development and production build process (straight forward process)
- Minor improvements to make it easier to write components in cdap react app.

### CODE REVIEW

**Comment by anwar6953** [comment]
just curious - what does 'single sign-on' mean exactly?

File: cdap-ui/Gulpfile.js [hunk 1]
```diff
@@ -24,10 +24,83 @@ var gulp = require('gulp'),
...
+    './app/filters/**/*.js',
+    './app/lib/global-lib.js',
+    './app/' + extension + '/module.js',
+    './app/' + extension + '/**/*.js',
+  ];
+  source = source.concat(getEs6Directives(true));
+
+  return gulp.src(source)
+    .pipe(plug.plumber())
+    .pipe(plug.ngAnnotate())
```
Comment (elfenheart) [inline cdap-ui/Gulpfile.js]: we should move this ngAnnotate to be after babel

File: cdap-ui/Gulpfile.js [hunk 2]
```diff
@@ -24,10 +24,83 @@ var gulp = require('gulp'),
...
+  var PKG = JSON.stringify({
+    name: pkg.name,
+    v: pkg.version
+  });
+
+  var source = getEs6Directives();
+
+  return gulp.src(source)
+    .pipe(plug.plumber())
+    .pipe(plug.ngAnnotate())
```
Comment (elfenheart) [inline cdap-ui/Gulpfile.js]: same here

File: cdap-ui/app/tracker/main.js [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+
+      'ncy-angular-breadcrumb',
+      'angularMoment',
+      'ui.ace',
+      'gridster',
+      'angular-cron-jobs',
+      'angularjs-dropdown-multiselect',
+      'hc.marked',
+      'ngFileSaver',
+      'infinite-scroll'
```
Comment (elfenheart) [inline cdap-ui/app/tracker/main.js:81]: might not come in this PR, but we should be able to limit these dependencies as well right?

File: cdap-ui/app/tracker/main.js [hunk 4]
```diff
@@ -0,0 +1,373 @@
...
+  .run(function ($rootScope, $state, $stateParams) {
+    // It's very handy to add references to $state and $stateParams to the $rootScope
+    // so that you can access them from any scope within your applications.For example,
+    // <li ng-class="{ active: $state.includes('contacts.list') }"> will set the <li>
+    // to active whenever 'contacts.list' or one of its decendents is active.
+    $rootScope.$state = $state;
+    $rootScope.$stateParams = $stateParams;
+
+    // for debugging... or to trigger easter eggs?
+    window.$go = $state.go;
```
Comment (elfenheart) [inline cdap-ui/app/tracker/main.js:96]: why not just attach entire $state to window? then we can get state params as well

File: cdap-ui/app/tracker/main.js (hunk 3 above)
Comment (ajainarayanan) [reply cdap-ui/app/tracker/main.js:81]: Good point. Will fix shouldn't take too much time.

File: cdap-ui/app/tracker/main.js (hunk 4 above)
Comment (ajainarayanan) [reply cdap-ui/app/tracker/main.js:96]: This should go away. This should probably not be here. Storing or having a reference of something inside angular in window might not be a good idea. Good catch. Will remove it.

**Comment by ajainarayanan** [comment]
Bamboo build: http://builds.cask.co/browse/CDAP-DRC4293-15

**Comment by ajainarayanan** [comment]
Talked to @sreevatsanraman about on-hold. We are good to merge for now.

**Pull request merged**